test/libsolidity/SemanticTest.cpp

```diff
@@ -92,18 +91,33 @@ SemanticTest::SemanticTest(
 		m_shouldRun = false;
 
 	auto const eofEnabled = solidity::test::CommonOptions::get().eofVersion().has_value();
-	std::string compileViaYul = m_reader.stringSetting("compileViaYul", eofEnabled ? "true" : "also");
 
-	if (compileViaYul == "false" && eofEnabled)
+	auto const compileViaYul = m_reader.enumSetting<CompileViaYul>(
+		"compileViaYul",
+		{
+			{toString(CompileViaYul::False), CompileViaYul::False},
+			{toString(CompileViaYul::True), CompileViaYul::True},
+			{toString(CompileViaYul::Also), CompileViaYul::Also},
+			{toString(CompileViaYul::OnlyOnEOF), CompileViaYul::OnlyOnEOF}
+		},
+		eofEnabled ? toString(CompileViaYul::True) : toString(CompileViaYul::Also)
+	);
+
+	if (compileViaYul == CompileViaYul::OnlyOnEOF)
+		solUnimplemented("'compileViaYul: onlyOnEOF' is not yet supported in semantic tests.");
+
+	if (bytecodeFormat().contains(BytecodeFormat::EOFv1) && compileViaYul == CompileViaYul::False)
+		BOOST_THROW_EXCEPTION(std::runtime_error("Compilation to EOF requires using Yul IR"));
+
+	if (compileViaYul == CompileViaYul::False && eofEnabled)
 		m_shouldRun = false;
```

Comment:
The `eofEnabled` check should be converted to an assert, like in the syntax test. And we adjust the message the same way:

```suggestion
	if (m_compileViaYul == CompileViaYul::False)
	{
		soltestAssert(!eofEnabled);
		if (bytecodeFormat().contains(BytecodeFormat::EOFv1))
			BOOST_THROW_EXCEPTION(std::runtime_error(
				"Conflicting test settings: 'bytecodeFormat' includes EOF, which requires compilation via IR, but 'compileViaYul' is set to false."
			));
	}
```